### ISSUE TITLE
Discrepancy between Match-or-Conf description and implementation

### ISSUE THREAD

**Issue by Surpass-happy** [reddyav1/unite#6]
Hi @reddyav1,
In the paper, you described Match-or-Conf as a collaborative selection. However, in run_stage3.py, I noticed:
most_conf_preds = torch.where(student_conf, preds_full_t.detach(), clip_preds)
is immediately followed by:
most_conf_preds = preds_full_t.detach()

This effectively overwrites the CLIP teacher's contribution. Is this "student-only" pseudo-labeling intentional for better convergence? Why not use the collaborative labels from the first line?
